From 58ed20ef1d98e60569e7766d440b5f0ff7fc47f4 Mon Sep 17 00:00:00 2001
From: Steve Cornelius <steve.cornelius@freescale.com>
Date: Fri, 23 Mar 2012 10:00:00 -0700
Subject: ENGR00180943-5: Add CAAM instantiation data to i.MX6 common platform
 device set

Add CAAM instantiation data to i.MX6 common platform device set.

Signed-off-by: Steve Cornelius <steve.cornelius@freescale.com>
---
 arch/arm/plat-mxc/include/mach/devices-common.h |   12 ++++++++++++
 arch/arm/plat-mxc/include/mach/mx6.h            |    2 ++
 2 files changed, 14 insertions(+)

diff --git a/arch/arm/plat-mxc/include/mach/devices-common.h b/arch/arm/plat-mxc/include/mach/devices-common.h
index 6fce8c4..4ae60dc 100755
--- a/arch/arm/plat-mxc/include/mach/devices-common.h
+++ b/arch/arm/plat-mxc/include/mach/devices-common.h
@@ -104,6 +104,18 @@ struct imx_snvs_rtc_data {
 struct platform_device *__init imx_add_snvs_rtc(
 		const struct imx_snvs_rtc_data *data);
 
+struct imx_caam_data {
+	resource_size_t iobase_caam;
+	resource_size_t irq_sec_vio;
+	resource_size_t irq_snvs;
+	resource_size_t irq_jr0;
+	resource_size_t irq_jr1;
+	resource_size_t irq_jr2;
+	resource_size_t irq_jr3;
+};
+
+struct platform_device *__init imx_add_caam(
+		const struct imx_caam_data *data);
 
 #include <mach/imxfb.h>
 struct imx_imx_fb_data {
diff --git a/arch/arm/plat-mxc/include/mach/mx6.h b/arch/arm/plat-mxc/include/mach/mx6.h
index 67d00df..79093aa 100644
--- a/arch/arm/plat-mxc/include/mach/mx6.h
+++ b/arch/arm/plat-mxc/include/mach/mx6.h
@@ -209,6 +209,8 @@
 #define ENET_BASE_ADDR			(AIPS2_OFF_BASE_ADDR+0x8000)
 #define MLB_BASE_ADDR			(AIPS2_OFF_BASE_ADDR+0xC000)
 
+#define MXC_CAAM_BASE_ADDR		(AIPS2_ARB_BASE_ADDR + 0x00000)
+
 #define MX6Q_USDHC1_BASE_ADDR		(AIPS2_OFF_BASE_ADDR + 0x10000)
 #define MX6Q_USDHC2_BASE_ADDR		(AIPS2_OFF_BASE_ADDR + 0x14000)
 #define MX6Q_USDHC3_BASE_ADDR		(AIPS2_OFF_BASE_ADDR + 0x18000)
-- 
1.7.9.5

